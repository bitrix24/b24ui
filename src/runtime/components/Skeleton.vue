<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/skeleton'
import type { ComponentConfig } from '../types/utils'

type Skeleton = ComponentConfig<typeof theme, AppConfig, 'skeleton'>

export interface SkeletonProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  class?: any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { tv } from '../utils/tv'

const props = defineProps<SkeletonProps>()

const appConfig = useAppConfig() as Skeleton['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.skeleton || {}) }))
</script>

<template>
  <Primitive :as="as" :class="b24ui({ class: props.class })">
    <slot />
  </Primitive>
</template>
