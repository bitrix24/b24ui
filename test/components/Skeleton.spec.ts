import { describe, it, expect } from 'vitest'
import Skeleton from '../../src/runtime/components/Skeleton.vue'
import type { SkeletonProps } from '../../src/runtime/components/Skeleton.vue'
import ComponentRender from '../component-render'

describe('Skeleton', () => {
  it.each([
    // Props
    ['with as', { props: { as: 'span' } }],
    ['with class', { props: { class: 'rounded-full size-12' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: SkeletonProps }) => {
    const html = await ComponentRender(nameOrHtml, options, Skeleton)
    expect(html).toMatchSnapshot()
  })
})
