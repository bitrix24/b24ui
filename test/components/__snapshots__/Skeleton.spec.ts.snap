// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > renders with as correctly 1`] = `"<span class="animate-pulse rounded-md bg-gray-200 dark:bg-gray-800"></span>"`;

exports[`Skeleton > renders with class correctly 1`] = `"<div class="animate-pulse bg-gray-200 dark:bg-gray-800 rounded-full size-12"></div>"`;
